daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -661,14 +658,13 @@ private[speedy] case class PartialTransaction(
   def endWithAuthority: PartialTransaction = {
     context.info match {
       case info: WithAuthorityContextInfo =>
-        val was = info.authorizers
-        val now = info.parent.info.authorizers
-        println(s"**endWithAuthority\n- was=$was\n- now=$now") // TODO #15882: remove debug
+        val authNode = Node.Authority(
+          obtained = info.authorizers,
+          children = context.children.toImmArray,
+        )
         copy(
-          context = info.parent.copy(
-            children = info.parent.children :++ context.children.toImmArray,
-            nextActionChildIdx = context.nextActionChildIdx,
-          )
+          context = info.parent.addActionChild(info.nodeId, context.minChildVersion),
```

Comment:
Why do we handle this node differently that Rollback ? 
(1) authority node are not an action node, and (2) they do not have version.

Reply:
So we should use `addRollbackChild` (perhaps renamed as `addNonActionChild`)?

````
    def addActionChild(child: NodeId, version: TxVersion): Context = {
      Context(info, minChildVersion min version, children :+ child, nextActionChildIdx + 1)
    }
    def addRollbackChild(child: NodeId, version: TxVersion, nextActionChildIdx: Int): Context =
      Context(info, minChildVersion min version, children :+ child, nextActionChildIdx)
```

Question: Why are we tracking `nextActionChildIdx: Int` in `Context` anyway?
Where is it used?